app/src/main/java/org/groundplatform/android/model/imagery/TileSource.kt

```diff
@@ -17,9 +17,37 @@ package org.groundplatform.android.model.imagery
 
 import org.groundplatform.android.model.map.Bounds
 
-/** Represents a single source of tiled map imagery. */
+/**
+ * Represents a single source of tiled map imagery used for displaying maps in the application.
+ *
+ * A [TileSource] defines where and how the map tiles are fetched — either from local offline
+ * storage or from a remote source (e.g., MOG or online tile servers).
+ */
 sealed class TileSource
 
-data class LocalTileSource(val localFilePath: String, val clipBounds: List<Bounds>) : TileSource()
+/**
+ * Represents a locally stored set of map tiles, typically used for offline map rendering.
+ *
+ * These tiles are fetched from the device’s local storage and may be limited by bounding
+ * coordinates and zoom levels.
+ *
+ * @property localFilePath The absolute file path on device storage where the local tile files are
+ *   stored.
+ * @property clipBounds The list of geographic bounding boxes (in lat/lng) within which these tiles
+ *   are valid.
+ * @property maxZoom The maximum zoom level supported by this offline tile source.
+ */
+data class LocalTileSource(
+  val localFilePath: String,
+  val clipBounds: List<Bounds>,
+  val maxZoom: Int,
+) : TileSource()
 
+/**
+ * Represents a remote map tile source hosted on a server, such as MOG (Map Over the Grid).
```

Comment:
MOG doesn't stand for "Map Over the Grid". It refers to "Google Maps Optimized GeoTIFFs"